### DISCUSSION TITLE
Question about Tracer Provider Versions

### DISCUSSION THREAD

**Discussion by thetizzo** [open-telemetry/opentelemetry-ruby#1040] [Q&A] [answered]
Hi 👋 

I've seen in a lot of the documentation and examples, the trace provider being called like this:

```ruby
tracer = OpenTelemetry.tracer_provider.tracer('my-app', '0.1.0')
```

My question is about the version number.  If the tracer that I want to get a handle on is one from the auto-instrumentation, Sidekiq for example, then I can do something like this: 

```ruby
tracer = OpenTelemetry.tracer_provider.tracer("OpenTelemetry::Instrumentation::Sidekiq", <version>)
```

However the issue with this is that I then have to keep this call up to date whenever the version number of the gem changes or I am potentially creating a new tracer instance with different than intended version here: https://github.com/open-telemetry/opentelemetry-ruby/blob/f08c802f726b792c029d4f50a878dba69398ff56/sdk/lib/opentelemetry/sdk/trace/tracer_provider.rb#L55  Is there a recommended way to programatically find the version number for a tracer like this that will keep this updated as the gem gets updated?  

Should I only be using the tracer provider to find my own custom tracers and not use the auto-instrumenter ones directly?

**Comment by arielvalentin** [comment, marked as answer]
> Should I only be using the tracer provider to find my own custom tracers and not sure the auto-instrumenter ones directly?

Yes, you should create as many named tracers as your application needs and avoid using the tracers created by auto-instrumented libraries. The auto-instrumentation libraries register themselves with the appropriate version when they are installed so there is nothing to be concerned about there.  

The `name` and `version` parameters are mapped to [OTLP `InstrumentationLibrary.name` and `Instrumentation.version`](https://github.com/open-telemetry/opentelemetry-proto/blob/b43e9b18b76abf3ee040164b55b9c355217151f3/opentelemetry/proto/common/v1/common.proto#L77) when the data is exported to OTLP collectors. 

Interpreting these attributes is vendor specific, e.g. [the OTEL Collector maps them as span attributes when exporting them to Jaeger](https://github.com/open-telemetry/opentelemetry-go/blob/main/exporters/jaeger/jaeger.go#L34).

- https://opentelemetry.io/docs/reference/specification/trace/api/#get-a-tracer
- https://opentelemetry.io/docs/concepts/instrumenting/#configure-the-opentelemetry-api

HTH
